### PR TITLE
{exporter/datadog,processor/datadog}: bump datadog-agent dependencies

### PR DESCRIPTION
**Pull request by gbbr**

### CODE REVIEW

File: cmd/configschema/go.mod [hunk 1]
```diff
@@ -16,6 +16,8 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
 )
 
+require github.com/DataDog/opentelemetry-mapping-go/pkg/quantile v0.1.0 // indirect
```
Comment (mx-psi) [inline cmd/configschema/go.mod:19]: Can you move this together with all the other `// indirect` dependencies?

File: cmd/otelcontribcol/go.mod [hunk 2]
```diff
@@ -571,8 +572,8 @@ require (
 	go.opentelemetry.io/collector/confmap v0.70.1-0.20230130215412-26bd7b2bf678 // indirect
 	go.opentelemetry.io/collector/consumer v0.70.1-0.20230130215412-26bd7b2bf678 // indirect
 	go.opentelemetry.io/collector/featuregate v0.70.1-0.20230130215412-26bd7b2bf678 // indirect
-	go.opentelemetry.io/collector/pdata v1.0.0-rc4.0.20230130215412-26bd7b2bf678 // indirect
-	go.opentelemetry.io/collector/semconv v0.70.1-0.20230130215412-26bd7b2bf678 // indirect
+	go.opentelemetry.io/collector/pdata v1.0.0-rc5 // indirect
+	go.opentelemetry.io/collector/semconv v0.71.0 // indirect
```
Comment (mx-psi) [inline cmd/otelcontribcol/go.mod:576]: This is already updated on main by fac9f8ba05b3af51e4e11e2cb639db3a4b7202d0, you need to rebase

File: cmd/configschema/go.mod (hunk 1 above)
Comment (mx-psi) [reply cmd/configschema/go.mod:19]: Actually, I think this will go away with the rebase

File: cmd/otelcontribcol/go.mod (hunk 2 above)
Comment (gbbr) [reply cmd/otelcontribcol/go.mod:576]: Oh my, I was way far behind. Sorry about that.

**Pull request closed without merging**